packages/react-dom-bindings/src/events/ReactDOMEventReplaying.js

```diff
@@ -329,36 +340,49 @@ function attemptExplicitHydrationTarget(
         }
       }
     }
+    queuedTarget.blockedOn = null;
+    const resolve = queuedTarget.resolve;
+    if (resolve) {
+      resolve();
+      queuedTarget.resolve = queuedTarget.reject = null;
+    }
   }
-  queuedTarget.blockedOn = null;
 }
 
-export function queueExplicitHydrationTarget(target: Node): void {
-  // TODO: This will read the priority if it's dispatched by the React
-  // event system but not native events. Should read window.event.type, like
-  // we do for updates (getCurrentEventPriority).
-  const updatePriority = getCurrentUpdatePriority();
-  const queuedTarget: QueuedHydrationTarget = {
-    blockedOn: null,
-    target: target,
-    priority: updatePriority,
-  };
-  let i = 0;
-  for (; i < queuedExplicitHydrationTargets.length; i++) {
-    // Stop once we hit the first target with lower priority than
-    if (
-      !isHigherEventPriority(
-        updatePriority,
-        queuedExplicitHydrationTargets[i].priority,
-      )
-    ) {
-      break;
+export function queueExplicitHydrationTarget(target: Node): Promise<void> {
+  return new Promise((resolve, reject) => {
```

Comment:
On thing that sucks here is that you're going to start getting unhandled promise rejections for these that used to be just noops that you can safely ignore.

I wonder if we should do a something to avoid this triggering unhandled promise rejections and that this can just safely be ignored.